generators/python_functions/soundex_generator/code_snippet.md

```diff
@@ -1,15 +1,14 @@
 ```python
 import unicodedata
-from pydantic import BaseModel
 
-YOUR_ATTRIBUTE = "word" # Choose any available attribute here.
+YOUR_ATTRIBUTE: str = "text" # Choose any available attribute here.
 
 def soundex_generator(record):
-    '''Converts an english word into a phonetic SoundEx representation, for example to store names.'''
+    """Converts an english word into a phonetic SoundEx representation, for example to store names."""
 
     word = record[YOUR_ATTRIBUTE].text # SpaCy doc, hence we need to use .text to get the string. 
```

Comment:
No, this is meant to be used on single words, such as surnames. As far as I understand, this is not meant to be used on sentences. To get the SoundEx for all words in a sentence, we could just split and iterate over a sentence?

Reply:
sounds good to me

Reply:
Hmmm, imo there is no reason for this to work because it would just give the soundex for the first word in the whole text. So we need to be careful here.